orangecontrib/network/widgets/tests/test_OWNxGroups.py

```diff
@@ -94,6 +96,55 @@ def test_send_report(self):
         self.send_signal(self.widget.Inputs.network, None)
         self.widget.send_report()
 
+    def test_weights(self):
+        label_var = DiscreteVariable("label", values=tuple("abcde"))
+        domain = Domain([label_var], [])
+        items = Table.from_numpy(domain, np.array([[0, 0, 1, 3, 4, 0, 1, 2, 4]]).T)
+        src, dst, weights = np.array(
+            [[0, 1, 5],
+             [1, 2, 4],
+             [3, 4, 6],
+             [0, 5, 3],
+             [1, 6, 1],
+             [2, 7, 2],
+             [3, 8, 8],
+             [4, 8, 7],
+             [5, 6, 2]]).T
+        edges = sp.coo_matrix((weights.astype(float), (src, dst)), shape=(9, 9))
+        network = Network(items, edges)
+        expected = np.zeros((5, 5), dtype=float)
+
+        widget = self.widget
+        self.send_signal(widget.Inputs.network, network)
+
+        widget.weighting = widget.NoWeights
```

Comment:
Usually I prefer simulating the click as much as possible:
`widget.controls.weighting.buttons[widget.NoWeights].click()`